hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -65,10 +65,10 @@ public void preHandle(@NonNull final PreHandleContext context) {
             // If a spender who is granted approveForAll from owner and is granting
             // allowance for a serial to another spender, need signature from the approveForAll
             // spender
-            var operatorId = allowance.delegatingSpenderOrElse(ownerId);
+            var operatorId = allowance.hasDelegatingSpender() ? allowance.delegatingSpender() : ownerId;
```

Comment:
Why the choice to make this change? It seems like in this case, the previous code would be preferable (no need for the ternary)

Reply:
The other problem here is the static analyzer cannot determine that operatorId is not sometimes null (it thinks `delegatingSpender` here might be null). So that makes line 77 get marked with a warning.

Reply:
Thanks Richard. `delegatingSpenderOrElse(ownerId)` makes sense. Reverting the change.